lib/chef/provider/package/rpm.rb

```diff
@@ -108,6 +108,14 @@ def remove_package(name, version)
           end
         end
 
+        private
+
+        def uri_scheme?(str)
+          scheme = URI.split(str).first
+          %w(http https ftp file).include?(scheme)
```

Comment:
Good call - The http RFC specifies case insensitivity, so we should do that here. Related: #2500